src/pages/profiles/add-profile-dialog.tsx

```diff
@@ -0,0 +1,129 @@
+/**
+ * Copyright (c) 2024, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+import React, { RefObject, useCallback } from 'react';
+import {
+    Button,
+    Dialog,
+    DialogActions,
+    DialogContent,
+    DialogContentText,
+    DialogTitle,
+    InputAdornment,
+    TextField,
+} from '@mui/material';
+import { FormattedMessage } from 'react-intl';
+import { Controller, SubmitHandler, useForm } from 'react-hook-form';
+import { ManageAccounts } from '@mui/icons-material';
+import { UserAdminSrv, UserProfile } from '../../services';
+import { useSnackMessage } from '@gridsuite/commons-ui';
+import { GridTableRef } from '../../components/Grid';
+import PaperForm from '../common/paper-form';
+
+export interface AddProfileDialogProps {
+    gridRef: RefObject<GridTableRef<UserProfile>>;
+    open: boolean;
+    setOpen: (open: boolean) => void;
+}
+
+const AddProfileDialog: React.FunctionComponent<AddProfileDialogProps> = (
+    props
+) => {
+    const { snackError } = useSnackMessage();
+
+    const { handleSubmit, control, reset, clearErrors } = useForm<{
+        name: string;
+    }>({
+        defaultValues: { name: '' }, //need default not undefined value for html input, else react error at runtime
+    });
+
+    const addProfile = useCallback(
+        (name: string) => {
+            const profileData: UserProfile = {
+                name: name,
+            };
+            UserAdminSrv.addProfile(profileData)
+                .catch((error) =>
+                    snackError({
+                        messageTxt: error.message,
+                        headerId: 'profiles.table.error.add',
+                    })
+                )
+                .then(() => props.gridRef?.current?.context?.refresh?.());
+        },
+        [props.gridRef, snackError]
+    );
+
+    const handleClose = () => {
+        props.setOpen(false);
+        reset();
+        clearErrors();
+    };
+
+    const onSubmit: SubmitHandler<{ name: string }> = (data) => {
+        addProfile(data.name.trim());
+        handleClose();
+    };
```

Comment:
I'm stuck here because if I start using useCallback, I end up with missing deps with handleClose, and then reset/clearErrors which are not mine but come from useForm.

Reply:
Hum ok, but still you're using these values/callbacks, so I think it's ok to put them in dependencies even if it doesn't come form you, no ?

Reply:
done